### PR TITLE
Fixing URL Decoding Issue (#446)

### PR DESCRIPTION
**Pull request by mathewc**

### CODE REVIEW

File: src/WebJobs.Script.WebHost/WebScriptHostManager.cs [hunk 1]
```diff
@@ -111,16 +113,15 @@ public FunctionDescriptor GetHttpFunctionOrNull(Uri uri)
...
             if (HttpFunctions == null || HttpFunctions.Count == 0)
             {
                 return null;
             }
 
             // Parse the route (e.g. "api/myfunc") to get 'myfunc"
             // including any path after "api/"
-            string route = uri.AbsolutePath;
+            FunctionDescriptor function = null;
+            string route = HttpUtility.UrlDecode(uri.AbsolutePath);
```
Comment (mathewc) [inline src/WebJobs.Script.WebHost/WebScriptHostManager.cs]: This UrlDecode call is the fix

File: test/WebJobs.Script.Tests/WebScriptHostManagerTests.cs [hunk 2]
```diff
@@ -88,6 +90,32 @@ public void SecretFilesArePurgedOnStartup()
...
+            TestInvoker invoker = new TestInvoker();
+            Collection<ParameterDescriptor> parameters = new Collection<ParameterDescriptor>();
+            FunctionDescriptor function = new FunctionDescriptor("Foo Bar", invoker, metadata, parameters);
+            Collection<FunctionDescriptor> functions = new Collection<FunctionDescriptor>()
+            {
+                function
+            };
+            manager.InitializeHttpFunctions(functions);
+
+            Uri uri = new Uri("http://local/api/Foo Bar");
```
Comment (davidebbo) [inline test/WebJobs.Script.Tests/WebScriptHostManagerTests.cs]: Might as well make the test a bit crazier and add some accented characters like éà, and Chinese chars like 中國.

**Comment by davidebbo** [comment]
:shipit:

File: test/WebJobs.Script.Tests/WebScriptHostManagerTests.cs (hunk 2 above)
Comment (mathewc) [inline test/WebJobs.Script.Tests/WebScriptHostManagerTests.cs]: I added a test case for that.

**Pull request closed without merging**